### PR TITLE
[Modules] Updated ManagedServices RegistrationDefinitions to new dependency app…

### PR DESCRIPTION
**Pull request by MrMCake**
…roach

# Description

- Updated ManagedServices RegistrationDefinitions to new dependency app…

| Pipeline |
| - |
| [ManagedServices: RegistrationDefinitions](https://github.com/Azure/ResourceModules/actions/workflows/ms.managedservices.registrationdefinitions.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.ManagedServices/registrationDefinitions/.test/rg/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,56 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes')
```
Comment (MrMCake) [inline modules/Microsoft.ManagedServices/registrationDefinitions/.test/rg/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes')
```

File: modules/Microsoft.ManagedServices/registrationDefinitions/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        principalIdDisplayName: 'ResourceModules-Contributor'
+        roleDefinitionId: 'b24988ac-6180-42a0-ab88-20f7382dd24c'
+      }
+      {
+        principalId: '87813317-fb25-4c76-91fe-783af429d109'
+        principalIdDisplayName: 'ResourceModules-LHManagement'
+        roleDefinitionId: '91c1777a-f3dc-4fae-b103-61d183457e46'
+      }
+    ]
+    managedByTenantId: '195ee85d-2f10-4764-8352-a3c99aa772fb'
```
Comment (eriqua) [inline modules/Microsoft.ManagedServices/registrationDefinitions/.test/common/deploy.test.bicep:34]: This looks good overall from a conversion-to-new-dependencies-approach point of view, but I really don't see this sustainable going forward. Who's tenant is this and how is this resource working if deployed outside of our test environment? I'd suggest to discuss this more in detail with the broader group.

Comment (MrMCake) [reply modules/Microsoft.ManagedServices/registrationDefinitions/.test/common/deploy.test.bicep:34]: It's Marius's. If you have a suggestion where we can get another tenant from which is more sustainable - let me know :D

Comment (eriqua) [reply modules/Microsoft.ManagedServices/registrationDefinitions/.test/common/deploy.test.bicep:34]: We have an issue open for more than a year (#307) and I just thought about seizing the opportunity to bring that up again giving the review, before it gets forgotten again.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**